apps/ltn/src/neighbourhood.rs

```diff
@@ -190,6 +205,29 @@ impl Neighbourhood {
             }
         }
 
+        // Add every connected road into connected_exterior_roads
+        let mut exterior: Vec<RoadID> = Vec::new();
+        for r in [&self.perimeter_roads, &self.interior_roads]
+            .into_iter()
+            .flatten()
+        {
+            exterior.extend(possible_destination_roads(map, *r, None));
+        }
+
+        debug!("BUILDING CONNECTED_EXTERIOR_ROADS: exterior.len() = {}", exterior.len());
+        debug!("BUILDING CONNECTED_EXTERIOR_ROADS: perimeter_roads.len() = {}", &self.perimeter_roads.len());
+        debug!("BUILDING CONNECTED_EXTERIOR_ROADS: interior_roads.len() = {}", &self.interior_roads.len());
+
+        // TODO **Surely** there is some set-wise way to do this?
```

Comment:
What if you make `exterior` a set instead of vec? Then something like

`connected_exterior_roads = exterior.difference(&self.perimeter_roads).difference(&self.interior_roads)`
or `exterior.difference(self.perimeter_roads.union(&self.interior_roads))`

Reply:
Refactored in https://github.com/a-b-street/abstreet/commit/e795459762149b70badcec1144b598724350db69